Source/WebKit/UIProcess/API/glib/WebKitUserContentManager.cpp

```diff
@@ -224,12 +306,60 @@ void webkit_user_content_manager_remove_all_scripts(WebKitUserContentManager* ma
     manager->priv->userContentController->removeAllUserScripts();
 }
 
+/**
+ * webkit_user_script_message_get_message:
+ * @message: A #WebKitUserScriptMessage
+ *
+ * Return the message sent in by user.
+ *
+ * Returns: (transfer none): The value sent by JavaScript code to the
+ * script message handler.
+ *
+ * Since: 2.38
+ */
+JSCValue* webkit_user_script_message_get_message(WebKitUserScriptMessage* userScriptMessage)
+{
+    if (userScriptMessage->message == nullptr)
+        return jsc_value_new_undefined(jsc_value_new());
+
+    return userScriptMessage->message;
+}
+
+/**
+ * webkit_user_script_message_reply:
+ * @message: A #WebKitUserScriptMessage
+ * @reply_value: Reply value of the provided script
+ *
+ * Provide the reply to the user provided script.
+ *
+ * Since: 2.38
+ */
+void webkit_user_script_message_reply(WebKitUserScriptMessage* message, JSCValue* replyValue)
+{
+    message->replyValue = replyValue;
```

Comment:
Like you did above, we always add `g_return[_val]_if_fail()` calls at the top of public API functions to do some checks on the passed values. For example here we would want to check the following:

```cpp
    g_return_if_fail(message != nullptr);
    g_return_if_fail(JSC_IS_VALUE(replyValue));
```